src/soundio/soundmanager.cpp

```diff
@@ -123,29 +123,28 @@ SoundManager::~SoundManager() {
     delete m_pMasterAudioLatencyOverload;
 }
 
-QList<SoundDevice*> SoundManager::getDeviceList(
+QList<QSharedPointer<SoundDevice>> SoundManager::getDeviceList(
     QString filterAPI, bool bOutputDevices, bool bInputDevices) {
     //qDebug() << "SoundManager::getDeviceList";
 
     if (filterAPI == "None") {
-        QList<SoundDevice*> emptyList;
-        return emptyList;
+        return QList<QSharedPointer<SoundDevice>>();
     }
 
     // Create a list of sound devices filtered to match given API and
     // input/output.
-    QList<SoundDevice*> filteredDeviceList;
+    QList<QSharedPointer<SoundDevice>> filteredDeviceList;
 
-    foreach (SoundDevice* device, m_devices) {
+    for (const auto& pDevice: m_devices) {
```

Comment:
This range-based for loop will create a deep copy of m_devices since the member function and consequently the member variable is non-const. IMHO a _getter_-function should always be const and idempotent.